src/mpid/ch4/netmod/ofi/ofi_comm.c

```diff
@@ -64,5 +73,29 @@ int MPIDI_OFI_mpi_comm_set_hints(MPIR_Comm * comm, MPIR_Info * info)
 {
     int mpi_errno = MPI_SUCCESS;
 
+    if (comm) {
+        /* If the user set a multi-nic hint, but num_nics = 1, disable multi-nic optimizations. */
+        if (MPIDI_OFI_global.num_nics > 1) {
+            int was_enabled_striping = MPIDI_OFI_COMM(comm).enable_striping;
+
+            /* Check if we should use striping */
+            if (comm->hints[MPIR_COMM_HINT_ENABLE_STRIPING] != -1)
+                MPIDI_OFI_COMM(comm).enable_striping = comm->hints[MPIR_COMM_HINT_ENABLE_STRIPING];
+            else
+                MPIDI_OFI_COMM(comm).enable_striping = MPIR_CVAR_CH4_OFI_ENABLE_STRIPING;
+
+            /* If striping was on and we disabled it here, decrement the global counter. */
+            if (was_enabled_striping > 0 && MPIDI_OFI_COMM(comm).enable_striping == 0)
+                MPIDI_OFI_global.num_comms_enabled_striping--;
+            /* If striping was off and we enabled it here, increment the global counter. */
+            else if (was_enabled_striping <= 0 && MPIDI_OFI_COMM(comm).enable_striping != 0)
+                MPIDI_OFI_global.num_comms_enabled_striping++;
+
+            if (MPIDI_OFI_COMM(comm).enable_striping) {
+                MPIDI_OFI_global.stripe_threshold = MPIR_CVAR_CH4_OFI_STRIPING_THRESHOLD;
+            }
+        }
+    }
+
```

Comment:
Why can't we do this in `MPIDI_OFI_mpi_comm_commit_pre_hook`?

Reply:
That hook triggers when the communicator is created. This hook is for when the user calls `MPI_Comm_set_info`. We need to know every time that hint changes so we can turn the features on/off.

Reply:
I see. So `MPIDI_OFI_mpi_comm_commit_pre_hook` can call this routine rather than having a separate (duplicate) code, right?

Reply:
I don't think that's exactly the right way to go given that the functions are supposed to do different things, but we could move this code into a static internal function that works for both I think. I'll look at it this morning.

Reply:
I updated the PR to refactor this. Sorry this couldn't be in fixup commits, but it caused too many rebase problems. This only impacts these two commits:

* 2e812bb4e80f - ch4/ofi: Add hint/CVAR for multiplexing
* 8cb1df75c154 - ch4/ofi: Add info hint for striping